pkg/3scale/amp/operator/system_options_provider.go

```diff
@@ -378,11 +378,7 @@ func (s *SystemOptionsProvider) setBackendOptions() error {
 		return fmt.Errorf("'%s' field of '%s' secret must have 'scheme://user:password@host/path' format", component.BackendSecretBackendListenerServiceEndpointFieldName, component.BackendSecretBackendListenerSecretName)
 	}
 
-	if urlObj.Path == "" {
-		urlObj.Path = "/internal/"
-	}
-
-	s.options.BackendServiceEndpoint = urlObj.String()
+	s.options.CoreInternalAPIEndpoint = urlObj.JoinPath("/internal/").String()
```

Comment:
Currently the PR builds the `CoreInternalAPIEndpoint` here, but I am wondering if it could make sense to remove this completely, and instead have the operator only set the env vars for the route (`BACKEND_PUBLIC_URL`) and the internal service (e.g. call it `BACKEND_URL`), and append `/internal/` path in porta configuration (e.g. `"#{BACKEND_URL}/internal/"`)